Packs/Microsoft365Defender/Integrations/Microsoft365Defender/Microsoft365Defender_test.py

```diff
@@ -0,0 +1,103 @@
+    assert convert_incident_to_readable(raw_incident) == converted_incident
+
+
+def mock_client(mocker, function: str = None, http_response=None):
+    mocker.patch.object(demisto, 'getIntegrationContext',
+                        return_value={'current_refresh_token': 'refresh_token', 'access_token': 'access_token'})
+    client = Client(
+        app_id='app_id',
+        verify=False,
+        proxy=False
+    )
+    if http_response:
+        mocker.patch.object(client, function, return_value=http_response)
+    return client
+
+
+def check_api_response(results, results_mock):
+    assert results.outputs_prefix == results_mock['outputs_prefix']
+    assert results.outputs_key_field == results_mock['outputs_key_field']
+    assert results.readable_output == results_mock['readable_output']
+    assert results.outputs == results_mock['outputs']
+
+
+def test_microsoft_365_defender_incidents_list_command(mocker):
+    from Microsoft365Defender import microsoft_365_defender_incidents_list_command
+    client = mock_client(mocker, 'incidents_list', util_load_json('./test_data/incidents_list_response.json'))
+    results = microsoft_365_defender_incidents_list_command(client, {'limit': 10})
+    check_api_response(results, util_load_json('./test_data/incidents_list_results.json'))
+
+
+def test_microsoft_365_defender_incident_update_command(mocker):
+    from Microsoft365Defender import microsoft_365_defender_incident_update_command
+    client = mock_client(mocker, 'update_incident', util_load_json('./test_data/incident_update_response.json'))
+    args = {'id': '263', 'tags': 'test1,test2', 'status': 'Active', 'classification': 'Unknown',
+            'determination': 'Other'}
+    results = microsoft_365_defender_incident_update_command(client, args)
+    check_api_response(results, util_load_json('./test_data/incident_update_results.json'))
+
+
+def test_microsoft_365_defender_advanced_hunting_command(mocker):
+    from Microsoft365Defender import microsoft_365_defender_advanced_hunting_command
+    client = mock_client(mocker, 'advanced_hunting', util_load_json('./test_data/advanced_hunting_response.json'))
+    args = {'query': 'AlertInfo'}
+    results = microsoft_365_defender_advanced_hunting_command(client, args)
+    check_api_response(results, util_load_json('./test_data/advanced_hunting_results.json'))
+
+
+def fetch_check(mocker, client, last_run, first_fetch_time, fetch_limit, mock_results):
+    mocker.patch.object(demisto, 'getLastRun', return_value=last_run)
+    results = fetch_incidents(client, first_fetch_time, fetch_limit)
+    assert len(results) == len(mock_results)
+    for incident, mock_incident in zip(results, mock_results):
+        assert incident['name'] == mock_incident['name']
+        assert incident['occurred'] == mock_incident['occurred']
+        assert incident['rawJSON'] == mock_incident['rawJSON']
+
+
+def test_fetch_incidents(mocker):
```

Comment:
please add a test for the second fetch flow as well.

Reply:
This test covers 4 fetches in order to check that the internal queue is working as expected.

Added comments and changed variables' names.